python/ray/dashboard/modules/job/common.py

```diff
@@ -189,8 +197,25 @@ class JobInfoStorageClient:
     JOB_DATA_KEY_PREFIX = f"{ray_constants.RAY_INTERNAL_NAMESPACE_PREFIX}job_info_"
     JOB_DATA_KEY = f"{JOB_DATA_KEY_PREFIX}{{job_id}}"
 
-    def __init__(self, gcs_aio_client: GcsAioClient):
+    def __init__(self, gcs_aio_client: GcsAioClient, log_dir: Optional[str] = None):
```

Comment:
let's not allow log_dir = None case? 


```python3
  def __init__(self, gcs_aio_client: GcsAioClient, log_dir: str):
```

Reply:
I kept log_dir as an optional variable because the JobInfoStorageClient is called from quite a few places to get job info (no updates to the KV store so no export events are emitted). These callers don’t have a log_dir, so unless we have a global log dir that can be fetched making this variable non optional requires a refactor throughout the jobs code.

Reply:
let's rename the variable to export_event_log_dir_root? Something where it's clearly meant to control one specific thing.